### PR TITLE
Feature/24012 ees integration and ess clear cache endpoint authorization

### PR DESCRIPTION
**Pull request by Shirin14926**
Add AD authorization in webhook and fix unit test cases for authorization

### CODE REVIEW

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API/Controllers/EssWebhookController.cs [hunk 1]
```diff
@@ -45,23 +49,39 @@ public IActionResult NewFilesPublishedOptions()
...
+            };
+            bool isAzureB2C = azureAdB2CHelper.IsAzureB2CUser(azureAdB2C, GetCurrentCorrelationId());
+            if (isAzureB2C)
+            {
+                Logger.LogInformation(EventIds.ESSB2CUserValidationEvent.ToEventId(), "Event was triggered with invalid Azure AD token from Enterprise event for Clear Cache Search and Download Event for _X-Correlation-ID:{correlationId}", GetCurrentCorrelationId());
+                Logger.LogInformation(EventIds.ESSClearCacheSearchDownloadEventCompleted.ToEventId(), "Clear Cache Event completed as Azure AD Authentication failed with OK response and _X-Correlation-ID:{correlationId}", GetCurrentCorrelationId());
+                return GetCacheResponse();
+            }
+
             var eventGridEvent = JsonConvert.DeserializeObject<CustomEventGridEvent>(request.ToString());
```
Comment (barrie-cooper) [inline UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API/Controllers/EssWebhookController.cs:66]: This seems to satisfy the CodeQL scanner and I don't see any problems making the change:
```suggestion
            var eventGridEvent = new CustomEventGridEvent();
            JsonConvert.PopulateObject(request.ToString(), eventGridEvent);
```
What do you think?

Comment (Shirin14926) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API/Controllers/EssWebhookController.cs]: Yes, It did worked. Thanks

Comment (usha13529) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API/Controllers/EssWebhookController.cs]: Thanks Barrie for looking into CodeQL scanning issue.

**Review by usha13529 — APPROVED** [review]
LGTM

**Review by barrie-cooper — APPROVED** [review]
Thanks for changes, all good

**Pull request merged**